test/secure/Curve25519.test.ts

```diff
@@ -0,0 +1,297 @@
+import { describe, it } from 'node:test'
+import assert from 'node:assert'
+import crypto from 'node:crypto'
+import { generateKeyPair, sharedKey } from '../../src/Curve25519'
+
+describe('Curve25519', () => {
+	describe('generateKeyPair', () => {
+		it('should generate valid key pairs from seed', () => {
+			const seed = new Uint8Array(32).fill(1)
+			const keyPair = generateKeyPair(seed)
+
+			assert(keyPair.public instanceof Uint8Array)
+			assert(keyPair.private instanceof Uint8Array)
+			assert.strictEqual(keyPair.public.length, 32)
+			assert.strictEqual(keyPair.private.length, 32)
+
+			// Verify the private key format constraints
+			assert.strictEqual(keyPair.private[0] & 248, keyPair.private[0])
+			assert.strictEqual(
+				keyPair.private[31] & 127,
+				keyPair.private[31] & 127,
+			)
+			assert.strictEqual(keyPair.private[31] & 64, 64)
+
+			// Verify the public key format constraints
+			assert.strictEqual(keyPair.public[31] & 127, keyPair.public[31])
```

Comment:
wouldn't it make more sense that the two keys are actually valid by encrypting a value with the public key and try to decrypt it with the private? those asserts are not so useful